### ISSUE TITLE
where can i assign step in function lr_lambda of Class WramupLinearSchedule?

### ISSUE THREAD

**Issue by lsy641** [huggingface/transformers#1144]
As i know LambdaLR's get_lr function recieve last_epoch as parameter, so where does it get the step of current training?

class WarmupLinearSchedule(LambdaLR):
    """ Linear warmup and then linear decay.
        Linearly increases learning rate from 0 to 1 over `warmup_steps` training steps.
        Linearly decreases learning rate from 1. to 0. over remaining `t_total - warmup_steps` steps.
    """
    def __init__(self, optimizer, warmup_steps, t_total, last_epoch=-1):
        self.warmup_steps = warmup_steps
        self.t_total = t_total
        super(WarmupLinearSchedule, self).__init__(optimizer, self.lr_lambda, last_epoch=last_epoch)

    def lr_lambda(self, step):
        if step < self.warmup_steps:
            return float(step) / float(max(1, self.warmup_steps))
        return max(0.0, float(self.t_total - step) / float(max(1.0, self.t_total - self.warmup_steps)))

**Comment by LysandreJik** [comment]
Hi, to use a scheduler you have to tell it when to perform an optimization step, as detailed on the [pytorch documentation](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate). It increases the step by one every time you call `scheduler.step()`.
